openerp/addons/base/ir/ir_qweb/fields.py

```diff
@@ -324,26 +315,39 @@ def record_to_html(self, record, field_name, options, values=None):
         # (integer > 0) from the currency's rounding (a float generally < 1.0).
         fmt = "%.{0}f".format(display_currency.decimal_places)
 
-        from_amount = record[field_name]
-
         if options.get('from_currency'):
-            from_amount = options['from_currency'].compute(from_amount, display_currency)
+            value = options['from_currency'].compute(value, display_currency)
 
         lang = self.user_lang()
-        formatted_amount = lang.format(fmt, display_currency.round(from_amount),
+        formatted_amount = lang.format(fmt, display_currency.round(value),
                                 grouping=True, monetary=True).replace(r' ', u'\N{NO-BREAK SPACE}')
 
         pre = post = u''
         if display_currency.position == 'before':
-            pre = u'{symbol}\N{NO-BREAK SPACE}'
+            pre = u'{symbol}\N{NO-BREAK SPACE}'.format(symbol=display_currency.symbol)
         else:
-            post = u'\N{NO-BREAK SPACE}{symbol}'
+            post = u'\N{NO-BREAK SPACE}{symbol}'.format(symbol=display_currency.symbol)
 
-        return u'{pre}<span class="oe_currency_value">{0}</span>{post}'.format(
-            formatted_amount, pre=pre, post=post,
-        ).format(
-            symbol=display_currency.symbol,
-        )
+        return u'{pre}<span class="oe_currency_value">{0}</span>{post}'.format(formatted_amount, pre=pre, post=post)
+
+    @api.model
+    def record_to_html(self, record, field_name, options, values=None):
+        options = dict(options)
+        # for backward compatibility to remove after v10
+        if isinstance(options.get('display_currency'), str):
+            _logger.warning("Use new syntax for '%s' monetary widget 'display_currency' option (instead of deprecated JSON syntax) to improve speed." % options['display_currency'])
+            options['display_currency'] = safe_eval.safe_eval(options['display_currency'], mode='eval', globals_dict=values)
+        if isinstance(options.get('from_currency'), str):
+            _logger.warning("Use new syntax for '%s' monetary widget 'from_currency' option (instead of deprecated JSON syntax) to improve speed." % options['from_currency'])
+            options['from_currency'] = safe_eval.safe_eval(options['from_currency'], mode='eval', globals_dict=values)
+        # end backward
```

Comment:
Oups, I forget that 'value_to_html' does not have 'values' as attributes, I can't move this part :-/